tools/hilbert.py

```diff
@@ -0,0 +1,589 @@
+
+    args = parser.parse_args(args)
+    _args = vars(args)
+    assert 'station' in _args
+    stationId = _args['station']
+
+    action_args = None
+    if 'action_args' in _args:
+        action_args = _args['action_args']
+
+    cmd_action(parser, context, args, stationId, 'finish', action_args)
+
+    log.debug("Done")
+    exit(0)
+
+
+@subcmd('app_switch', help='Load Configuration and app_switch top application to the given one on one of the stations')
+def cmd_app_switch(parser, context, args):
+    log.debug("Running '{}'" . format('cmd_app_switch'))
+
+    parser.add_argument('-s', '--station', required=True, help="specify the station")
+    parser.add_argument('-g', '--action_args', required=False, help="specify the action arguments (new ApplicationID)")
+
+    args = parser.parse_args(args)
+    _args = vars(args)
+    assert 'station' in _args
+    stationId = _args['station']
+
+    action_args = None
+    if 'action_args' in _args:
+        action_args = _args['action_args']
+
+    cmd_action(parser, context, args, stationId, 'app_switch', action_args)
+
+    log.debug("Done")
+    exit(0)
+
+
+
+@subcmd('station_action', help='Load Configuration and perform some action with one of stations')
+def cmd_station_action(parser, context, args):
+    log.debug("Running '{}'" . format('station_action'))
+
+    parser.add_argument('-s', '--station', required=True, help="specify the station")
+    parser.add_argument('-a', '--action', required=True, help="specify the action")
+    parser.add_argument('-g', '--action_args', required=False, help="specify the action arguments")
+
+    args = parser.parse_args(args)
+    _args = vars(args)
+
+    assert 'station' in _args
+    assert 'action' in _args
+
+    stationId = _args['station']
+    action = _args['action']
+
+    action_args = None
+    if 'action_args' in _args:
+        action_args = _args['action_args']
+
+    cmd_action(parser, context, args, stationId, action, action_args)
+
+    log.debug("Done")
+    exit(0)
+
+
+def main():
+    handler = ArgumentHandler(use_subcommand_help=True, enable_autocompletion=True, description="Hilbert - server tool")
+
+    # add_argument, set_logging_level, set_subcommands,
+#    handler.add_argument('-q', '--quiet', help='decrease verbosity', action='store_true')
+#    handler.add_argument('-v', '--verbose', help='increase verbosity', action='store_true')
+
+#    handler.set_logging_argument('-l', '--log_level', default_level=logging.INFO)
+
+    # no arguments: 'store_const', 'store_true' or 'store_false'
+    handler.add_argument('-p', '--pedantic', required=False, action='store_true',
+                         help="turn on pedantic mode")
+
+    # 2 input sources: .YAML or .PICKLE
+    handler.add_argument('-if', '--inputfile', required=False, 
+                         help="specify input .YAML file (default: 'Hilbert.yml')")
+    handler.add_argument('-id', '--inputdump', required=False,
```

Comment:
High level representations of data (e.g. a JSON file or SQL file) are also called dumps... in fact, if you have a database in binary format the dump is a SQL text file... so calling the "compiled" representation a dump is confusing.

In NORMAL operation the UI will be using one cfg, and the cli should use the same... and we said the UI server should restart if the cfg changes... so users should be careful not to use the cli with the wrong cfg. Having the option of running the "pickle" file or the YAML one seems error prone and "dangerous" in this way. I recommend the YAML and pickle dates be compared, if the YAML is newer it's automatically compiled to .pickle in the same location. If not, the pickle is used directly.

... But really... it's an optimization already where the program doesn't even do anything. We shouldn't optimize before we even have code.